library/kotlin/io/envoyproxy/envoymobile/EngineBuilder.kt

```diff
@@ -26,6 +26,7 @@ open class EngineBuilder(
     EnvoyEngineImpl(onEngineRunning, logger, eventTracker)
   }
   private var logLevel = LogLevel.INFO
+  private var componentLogLevel = ""
```

Comment:
What are the options for component log level? If it's a finite set of strings, would making an enum configuration make more sense?

Reply:
It's a string of the format `<component>:<level>,<comopnent>:<level>`, and the components needs to be statically registered in Envoy. I started this with a map, but was deterred by the fact that passing a map through the C layer is painful, and decomposing the map into the string in the platform layer felt repetitive.

Open to suggestions though :)

Reply:
The current form does feel a little magical. If we don't want to do a map of `component (string): LogLevel (enum)`, let's at least document the expected structure inline and in the documentation for the builder

Reply:
I'm fine with the proposed changes but I would like to understand where these values are registered in Envoy. 

We can fixate those values on the platform layer which would make the validation in the common layer more sane while providing a clear set of values for the API.

Reply:
The components are statically registered here: https://github.com/envoyproxy/envoy/blob/main/source/common/common/logger.h#L29-L79

If we wanted to we could add a C API that let us query https://github.com/envoyproxy/envoy/blob/main/source/common/common/logger.h#L312 in order to get the list of registered log names. Another option would be to just capture a static list of valid log components that we manually keep in sync.

LMK which option ya'll prefer between the above and just documenting the expected log component string, happy to do whichever

Reply:
@buildbreaker ping on this, lmk how you want to proceed

Reply:
I think posting the above link to the components and having an example string in the public API comments/docstrings would be helpful.